Comments Added

MilanDonhowe opened the pull request:
### I added some simple comments to try and explain the functionality of the python script.
I have never seen "bunnybot" before so most of the functionality of the script was inferred, if I was incorrect in any of the commented statements I would be happy to correct them.

MilanDonhowe committed:
Comments Added

I added some simple comments to try and explain the functionality of the python script.
I have never seen "bunnybot" before so most of the functionality of the script was inferred, if I was incorrect in any of the commented statements I would be happy to correct them.

MilanDonhowe committed:
Removed pesky whitespace.

I accidentally added some white space to the file which I believed resulted in errors.

MilanDonhowe committed:
Indentation fix?

AHHHHHHHH

MilanDonhowe committed:
Indentation fixed for real

Now it should work.

brendanburkhart commented on components/bunnygrabber.py:
```diff
@@ -1,19 +1,31 @@
+# import Robot py library so this file can interface with the robot.
```
I think documenting our imports is unnecessary, this comment should be removed

brendanburkhart commented on components/bunnygrabber.py:
```diff
@@ -1,19 +1,31 @@
 class BunnyGrabber:
-    """Controls the bunny grabber.
-    """
+#	This program controls the bunny grabber arm.
```
We should be using Python [docstrings](https://www.python.org/dev/peps/pep-0257/) for documenting classes and functions, not comments

brendanburkhart commented on components/bunnygrabber.py:
```diff
@@ -1,19 +1,31 @@
 import wpilib
 
-
+# Class for controlling arm, will be instantiated in later script usage.
```
This doesn't control the whole arm, just the grabber at the end. This should become part of the classes docstring directly after the start of the class, also the comment about instantiating the class is unneeded.

brendanburkhart commented on components/bunnygrabber.py:
```diff
@@ -1,19 +1,31 @@
     direction = 0
 
+
+#	The following methods set_backward() and set_forward() tells the motor that
```
This comment should be split and added as a docstring (using `"""` triple quotes on each end) to each method individually.

brendanburkhart commented on components/bunnygrabber.py:
```diff
@@ -1,19 +1,31 @@
     def set_forward(self):
         self.direction = 1
 
+#   Method which allows the motor to move (or stop) based on if a direction has been specified with the previous methods.
```
This should also become a docstring right after the start of the method.

brendanburkhart commented on components/bunnygrabber.py:
```diff
@@ -1,19 +1,31 @@
 
+#   Method which allows the motor to move (or stop) based on if a direction has been specified with the previous methods.
     def execute(self):
+
```
This should be condensed a bit, into something like "Sets motor direction based off the `direction` variable"

MilanDonhowe committed:
Merge branch 'master' into MD-patch-1

MilanDonhowe committed:
Condensation.

I changed the comments to docstrings and moved their position and summarized the functionality of the script.

calebeby commented on components/bunnygrabber.py:
```diff
@@ -8,12 +8,21 @@ class BunnyGrabber:
         self.direction = -1
 
     def set_forward(self):
+        """ Sets motor direction to forwards.
```
Since this is a single-line docstring, the end quotes should be on the same line. Try to follow https://www.python.org/dev/peps/pep-0257/

calebeby replied:
Also no space before/after

calebeby replied:
This is not a useful comment.

calebeby commented on components/bunnygrabber.py:
```diff
@@ -8,12 +8,21 @@ class BunnyGrabber:
 
     def set_backward(self):
+        """
+            Sets motor direction to backwards.
```
This is not a useful comment. It does not say anything that the code does not.

brendanburkhart commented on components/bunnygrabber.py:
```diff
@@ -8,12 +8,21 @@ class BunnyGrabber:
         self.direction = 1
 
     def execute(self):
+        """ Moves the motor based on the direction variable.
```
Anyone working with this code should already know what a components `execute` method does, or they should look it up. This behavior is essentially what every single execute does. This docstring should be split up and added as standard comments to the actual code within the `execute` method, `execute` doesn't need a docstring.

brendanburkhart committed:
Merge branch 'master' into MD-patch-1

MilanDonhowe commented:
Since the code is self-explanatory I suppose it doesn't require documentation.
Not sure why I was asked to document this script.
I'll try and close the pull request.

calebeby commented:
Yeah, sorry, most of the robot code is pretty self explanatory and doesn't need much documentation.
